doc/install/build_install.rst

```diff
@@ -22,34 +22,18 @@ Minimum requirements
 The non distributed (i.e. no MPI) version of Arbor can be compiled on Linux or OS X systems
 with very few tools.
 
```

Comment:
I suggest adding small snippets on how to build Arbor on these platforms
* Ubuntu LTS
* MacOS
* Piz Daint
* Juwels Booster

picked from above.

Reply:
Yes please, a build guide catered to  the supported platforms would be incredibly helpful.

Reply:
You want this in this PR? The `build_install.rst` could use some more cleaning up IMHO, but I didn't want to merge that sort of change in this PR.

Reply:
Fine with adding it there.